src/markdown.js

```diff
@@ -43,6 +43,50 @@ export default function (config, options = {}) {
 	if (options.footnotes !== false) {
 		md.use(markdownItFootnote, options.footnotes);
 
+		let allFootnotes = new Map(); // pageId → { label: footnote }
+
+		/**
+		 * Extract and store footnote texts from a given set of tokens for a specific page
+		 * so that footnotes can be referenced and/or displayed appropriately in different document parts.
+		 * For example, it allows the markdown-it-footnote plugin and Paged.js to process the same footnotes correctly.
+		 * @param {markdownIt.Token[]} tokens
+		 * @param {string} pageId
+		 */
+		function extractFootnoteTexts (tokens, pageId) {
+			let footnotes = allFootnotes.get(pageId) ?? {};
+
+			let label, done;
+			for (let token of tokens) {
+				if (token.type === "footnote_open") {
+					// We have a new footnote to process
+					done = false;
+					label = token.meta.label;
+					footnotes[label] = "";
+				}
+				else if (token.type === "footnote_close") {
+					done = true;
+				}
+				else if (done === false && token.content) {
+					// Tokens of all other types after “footnote_open” till “footnote_close” form a footnote text
+					footnotes[label] += token.content;
```

Comment:
Have you tested this with footnote text that contains other tokens?

Reply:
If you mean the case when a footnote text has markdown markup (parsed as separate tokens), then yes, I tested it. That's why we render the footnote text before adding it to the footnote ref:

<img width="732" alt="image" src="https://github.com/user-attachments/assets/d0215471-2daa-4df3-8d5b-d70a0046081a">

Did I get your question right?

Reply:
Yes! It's unfortunate we're rendering the same piece of content multiple times, but what can we do.